prime-router/metadata/fhir_mapping/hl7/datatypes/XPN/HumanName.yml

```diff
@@ -0,0 +1,208 @@
+resourceType: HumanName
+
+family:
+  type: STRING
+  valueOf: XPN.1.1
+  expressionType: HL7Spec
+_family:
+  expressionType: resource
+  valueOf: datatypes/FN/ExtensionHumanName-family
+  specs: XPN.1
+
+given:
+  generateList: true
+  expressionType: nested
+  expressions:
+    - expressionType: HL7Spec
+      valueOf: XPN.2
+      type: STRING
+    - expressionType: HL7Spec
+      valueOf: XPN.3
+      type: STRING
+suffix:
+  generateList: true
+  expressionType: nested
+  expressions:
+    - expressionType: HL7Spec
+      valueOf: XPN.4
+      type: STRING
+    - expressionType: HL7Spec
+      valueOf: XPN.6
+      type: STRING
+    - expressionType: HL7Spec
+      valueOf: XPN.14
+      type: STRING
+prefix:
+  type: STRING
+  valueOf: XPN.5
+  expressionType: HL7Spec
+use:
+  type: NAME_USE
+  valueOf: XPN.7
+  expressionType: HL7Spec
+
+# This needs to use variables because the message parses this field using NULLDT since it is deprecated which means that
+# the values do not get correctly loaded and will not be mapped over correctly using a spec.
+period_1:
+  vars:
+    xpn12: String, XPN.12
+    xpn13: String, XPN.13
+    start: String, XPN.10.1
+    end: String, XPN.10.2
+  condition: $xpn12 NULL && $xpn13 NULL
+  expressionType: resource
+  valueOf: datatypes/DR/Period
+
+extension_AssemblyOrder:
+  vars:
```

Comment:
All extensions have condition settings to prevent them from being created if there is no value to be stored. This prevents an empty object from being created if there is no data.